### PR TITLE
JBPM-8171 Stunner - adding documentation feature tests

### PR DESCRIPTION
**Pull request by tiagodolphine**
JBPM-7916 introduced process documentation feature, but this feature is almost not covered by tests, this PR is inserting tests it.

@romartin 
@LuboTerifaj

### CODE REVIEW

**Comment by tiagodolphine** [comment]
Jenkins retest this

**Comment by tiagodolphine** [comment]
Jenkins retest this

**Comment by tiagodolphine** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/components/glyph/ImageStripDOMGlyphRenderer.java [hunk 1]
```diff
@@ -83,11 +83,15 @@ public void destroy() {
...
+                                          final int clipY) {
         final FlowPanel panel = new FlowPanel();
         panel.addStyleName(className);
-        panel.getElement().setAttribute("style", "background-position: " + clipX + "px " + clipY + "px !important");
+        panel.getElement().setAttribute("style", backGroundPosition(clipX, clipY));
         return panel;
     }
-}
+
+    protected static String backGroundPosition(int clipX, int clipY) {
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/components/glyph/ImageStripDOMGlyphRenderer.java]: can be `private`?

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/test/java/org/kie/workbench/common/stunner/core/client/util/js/JsConverterTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import static org.junit.Assert.assertEquals;
+
+public class JsConverterTest {
+
+    @Test
+    public void fromMap() {
+        final Map<?, ?> map = new Maps.Builder<String, String>()
+                .put("name", "tiago")
+                .put("age", "34")
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/test/java/org/kie/workbench/common/stunner/core/client/util/js/JsConverterTest.java]: only 34? it looks a bit more... :P (joking)

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/components/glyph/ImageStripDOMGlyphRenderer.java (hunk 1 above)
Comment (tiagodolphine) [reply kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/components/glyph/ImageStripDOMGlyphRenderer.java]: I let it as protected to cover it with unit test

**Comment by tiagodolphine** [comment]
@romartin  @LuboTerifaj  what about this one? more comments?

**Comment by LuboTerifaj** [comment]
Jenkins execute full downstream build

**Comment by LuboTerifaj** [comment]
It is not possible to log in to the business-central. I need to check that Documentation is not broken.

**Comment by LuboTerifaj** [comment]
Jenkins execute full downstream build

**Comment by tiagodolphine** [comment]
Jenkins execute full downstream build

**Comment by tiagodolphine** [comment]
Hi @LuboTerifaj there was an issue with dependency injection while refactoring for the tests, my fault. Now it is fixed, thanks!

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/documentation/ClientBPMNDocumentationServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,575 @@
...
+    public static final String PROCESS_DOCUMENTATION = "Process Documentation\nLine2";
+    public static final String PROCESS_VERSION = "1.0";
+    public static final String PROCESS_DECRIPTION = "Process Description";
+    public static final String PROCESS_UUID = UUID.uuid();
+    public static final String TASK_NAME = "Task";
+    public static final String TASK_DOC = "Task Doc";
+    public static final String TASK_PRIORITY = "1";
+    public static final String TASK_SUBJECT = "Task Subject";
+    public static final String TASK_DESCRIPTION = "Task Description";
+    public static final String PACKAGE = "org.kie";
```
[the remark this answers is not in the document]
Comment (tiagodolphine) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/documentation/ClientBPMNDocumentationServiceTest.java]: Ok

**Comment by tiagodolphine** [comment]
Hi @LuboTerifaj about the tests, it is not necessary to test all nodes, because the implementation is generic, based on the adapters to get properties, and the definitions, and values  assigned to them, more details https://github.com/kiegroup/kie-wb-common/blob/e4d4be143ef276384932766c67788c7dca272f35/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/documentation/ClientBPMNDocumentationService.java there is no hardcoded Definitions or Properties, so what should be tested is the flow of the adapters calls per functionality, that set the proper values on the Documentation Model output . 
I made a test for subprocess to test the Variables inside the subprocess, that is a different flow from the Global Variables and the Process Variables. 

Regarding the formatting thing you mentioned, I didn't find any issue, I used the kie formatter to auto format the classes on the IDE, can you point on the code the issues you found?

Thanks !!!

**Comment by LuboTerifaj** [comment]
Jenkins retest please.

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/components/glyph/ImageStripDOMGlyphRenderer.java [hunk 4]
```diff
@@ -83,11 +83,15 @@ public void destroy() {
     }
 
     private static FlowPanel buildPanel(final String className,
-                                        final int clipX,
-                                        final int clipY) {
+                                          final int clipX,
```
Comment (LuboTerifaj) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/components/glyph/ImageStripDOMGlyphRenderer.java]: Redundant spaces here.

Comment (tiagodolphine) [reply kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/components/glyph/ImageStripDOMGlyphRenderer.java]: :+1:

**Comment by tiagodolphine** [comment]
Jenkins retest please.

**Comment by tiagodolphine** [comment]
@LuboTerifaj applied your comments.

**Comment by LuboTerifaj** [comment]
Jenkins retest please.

**Comment by LuboTerifaj** [comment]
Jenkins execute full downstream build

**Comment by tiagodolphine** [comment]
@romartin this is ok to be merged.

**Pull request merged**